### ISSUE TITLE
Yes, @dss010101, it is best practice to specify the filename for `dst`, not just a directory. The latter _may_ work, but you risk an error.

### ISSUE THREAD

**Issue by dss010101** [paramiko/paramiko#2282]
Yes, @dss010101, it is best practice to specify the filename for `dst`, not just a directory. The latter _may_ work, but you risk an error.

_Originally posted by @bskinn in https://github.com/paramiko/paramiko/issues/1000#issuecomment-1678375288_

It's fine if the answer is 'thats just how it works and we won't change it'.  But I don't agree that it is 'best practice '. Should work similar to what I can do with standard ftp/sftp..which is directory or directory + wildcard patterns is supported

**Comment by bskinn** [comment]
> But I don't agree that it is 'best practice '.

Fair enough! 😀 Perhaps "current least-worst practice" is a better phrase?

In any event, this definitely makes sense to me as a feature request. Labeled as such.
